src/components/QrCode.tsx

```diff
@@ -0,0 +1,24 @@
+import log from "loglevel";
+import QRCode from "qrcode/lib/server";
+import { createSignal } from "solid-js";
+
+import "../style/qrcode.scss";
+
+export const Qrcode = ({ data }) => {
+    const [dataUrl, setDataUrl] = createSignal("");
+
+    QRCode.toDataURL(data, { width: 300 })
+        .then(setDataUrl)
+        .catch((err: Error) => {
+            log.error("qr code generation error", err);
+        });
```

Comment:
This should be in a `createEffect` call to make sure it's updated when `data` changes

Reply:
Agree, an invoice is fix for one swap id. Not sure what you mean @michael1011 , please ack that this can be merged